internal/cli/command/bucket/create.go

```diff
@@ -0,0 +1,261 @@
+	flags.StringVarP(&o.storageAccount, "storage-account", "", "", "Storage account for the bucket (must be specified for Azure)")
+	flags.StringVarP(&o.resourceGroup, "resource-group", "", "", "Resource group for the bucket (must be specified for Azure)")
+
+	return cmd
+}
+
+func runCreate(banzaiCli cli.Cli, o createBucketsOptions) error {
+	var err error
+
+	defaultBucketName := fmt.Sprintf("bucket-%s", strconv.FormatInt(time.Now().UTC().Unix(), 10))
+	orgID := input.GetOrganization(banzaiCli)
+
+	if o.cloud == "" {
+		// Select cloud
+		o.cloud, err = input.AskCloud()
+		if err != nil {
+			return err
+		}
+	} else {
+		err = input.IsCloudProviderSupported(o.cloud)
+		if err != nil {
+			return err
+		}
+	}
+
+	if o.name == "" {
+		// Ask for bucket name
+		err = survey.AskOne(&survey.Input{Message: "Bucket name:", Default: defaultBucketName}, &o.name, input.BucketNameValidator(o.cloud))
+		if err != nil {
+			return emperror.Wrap(err, "failed to select name")
+		}
+	} else {
+		err = input.ValidateBucketName(o.cloud, o.name)
+		if err != nil {
+			return emperror.Wrap(err, "failed to validate bucket name")
+		}
+	}
+
+	if o.location == "" {
+		// Ask for location
+		o.location, err = input.AskLocation(banzaiCli, o.cloud)
+		if err != nil {
+			return err
+		}
+	} else {
+		err = input.IsLocationValid(banzaiCli, o.cloud, o.location)
+		if err != nil {
+			return err
+		}
+	}
+
+	if o.secretID == "" {
+		// Ask for secret
+		o.secretID, err = input.AskSecret(banzaiCli, orgID, o.cloud)
+		if err != nil {
+			return err
+		}
+	}
+
+	secret, _, err := banzaiCli.Client().SecretsApi.GetSecret(context.Background(), orgID, o.secretID)
+	if err != nil {
+		return emperror.Wrap(utils.ConvertError(errors.WithStack(err)), "could not get secret")
+	}
+
+	if secret.Type != o.cloud {
+		return errors.Errorf("missmatch secret type '%s' versus cloud '%s'", secret.Type, o.cloud)
+	}
+
+	if o.cloud == input.CloudProviderAzure {
+		if banzaiCli.Interactive() {
+			// Ask for storage account
+			err = survey.AskOne(&survey.Input{Message: "Storage account:", Default: o.storageAccount}, &o.storageAccount, nil)
+			if err != nil {
+				return emperror.Wrap(err, "failed to get storage account")
+			}
+
+			// Ask for resource group
+			o.resourceGroup, err = input.AskResourceGroup(banzaiCli, orgID, o.secretID, o.resourceGroup)
+			if err != nil {
+				return emperror.Wrap(err, "failed to select resource group")
+			}
+		} else {
+			err = input.IsResourceGroupValid(banzaiCli, orgID, o.secretID, o.resourceGroup)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	request := getCreateBucketRequest(o)
+	response, _, err := banzaiCli.Client().StorageApi.CreateObjectStoreBucket(context.Background(), orgID, request)
+	if err != nil {
+		return emperror.Wrap(utils.ConvertError(errors.WithStack(err)), "could not create bucket")
+	}
+
+	log.Infof("bucket create request accepted for %s on %s", response.Name, response.Cloud)
+
+	var done bool
```

Comment:
this could default to `!options.wait` like for cluster creation `  -w, --wait           Wait for cluster creation`

Reply:
👍